tools/k8s/devcluster.ci.tpl.yaml

```diff
@@ -0,0 +1,48 @@
+stages:
+  - db:
+      name: db
+
+  - master:
+      pre:
+        - sh: make -C tools prep-root
+      config_file:
+        db:
+          host: localhost
+          port: 5432
+          password: postgres
+          user: postgres
+          name: determined
+        checkpoint_storage:
+          type: shared_fs
+          host_path: /tmp
+          storage_path: determined-cp
+        log:
+          level: debug
+        root: tools/build
+        cache:
+          cache_dir: /tmp/determined-cache
+        launch_error: false
+        security:
+          tls:
+            cert: $MASTER_TLS_CERT
+            key: $MASTER_TLS_KEY
+        resource_manager:
+          type: kubernetes
+          namespace: $KUBERNETES_NAMESPACE
+          slot_type: "cpu"
+          default_aux_resource_pool: default
+          default_compute_resource_pool: default
+          _master_ip: $MASTER_IP_FROM_INTERNAL
+          _master_port: $MASTER_PORT_FROM_INTERNAL
+          _use_user_kubeconfig: true
```

Comment:
it's not clear from the config but you can only use union<master ip/port, explicit kubeconfig, in cluster kubeconfig>. all together just picks the first in the chain (happens to be explicit kubeconfig).